src/seedu/addressbook/commands/Command.java

```diff
@@ -36,6 +36,14 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
         return String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, personsDisplayed.size());
     }
 
+    /**
+     * Determines if command perform mutation on data
+     * @return false by default
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.